fsm.cpp

```diff
@@ -18,12 +20,29 @@ int state1_countdown_timer = 0;
 int state2_duration_timer = 0;
 int state3_stillness_timer = 0;
 
+std::queue<sensorData_t> sensorDataQueue;
+DoorSensor doorSensor = DoorSensor(DOOR_SENSOR_PIN);
+MotionSensor motionSensor = MotionSensor(MOTION_SENSOR_PIN);
```

Comment:
There is a note in the Arduino docs saying that the `volatile` keyword should be used for variables shared by the interrupt handlers and the main code: https://www.arduino.cc/reference/en/language/functions/external-interrupts/attachinterrupt/ . So maybe all of these need it. However, all the examples that I see in Reply #7 of http://gammon.com.au/interrupts use primitive types like `int` and `bool`. Whereas the ones that you have are objects, which means that they should just be pointers which won't actually change. So it might not apply here. More to investigate.

Reply:
I'd lean towards adding the `volatile` keyword just in case. It might propagate down to the primitives that it's made up of like it sort of describes here about structs: https://en.cppreference.com/w/c/language/volatile

Reply:
Removed `volatile`s because they are only on classes and this will not auto-magically make them threadsafe, unfortunately.